packages/components/src/modifiers/hds-code-editor/languages/sentinel.ts

```diff
@@ -0,0 +1,182 @@
+import type { StringStream } from '@codemirror/language';
+
+type Quote = '"' | "'";
+
+interface ParserConfig {
+  delimiters?: RegExp;
+  operators?: RegExp;
+}
+
+interface SentinelState {
+  tokenize: (stream: StringStream, state: SentinelState) => string | null;
+}
+
+function wordRegexp(words: string[]): RegExp {
+  return new RegExp('^((' + words.join(')|(') + '))\\b');
+}
+
+// logical operators
+const wordOperators = wordRegexp(['and', 'or', 'not']);
+
+// keywords
+const sentinelKeywords = [
+  'import',
+  'main',
+  'rule',
+  'precondition',
+  'if',
+  'else',
+  'for',
+  'while',
+  'break',
+  'continue',
+  'return',
+  'in',
+  'each',
```

Comment:
we seem to be missing a few keywords here
```suggestion
  'all',
  'any',
  'as',
  'break',
  'case',
  'continue',
  'default',
  'else',
  'empty',
  'filter',
  'for',
  'func',
  'if',
  'import',
  'map',
  'param',
  'return',
  'rule',
  'when',
```